src/proposals.rs

```diff
@@ -161,15 +185,21 @@ async fn poll_approved_proposals(
                     }
 
                     if found_proposal {
-                        state.last_processed_proposal_id += missing_proposals;
+                        if pending_proposals {
+                            last_checked_pending_id += missing_proposals;
+                        } else {
+                            state.last_processed_proposal_id += missing_proposals;
+                        }
+
+                        continue;
```

Comment:
This `continue` addition is new, what's the benefit restarting the loop here brings?

Reply:
We want to keep processing if we find a new proposal with ID gaps, and break/wait otherwise. Previously we were always breaking instead of continuing if we found new ones, which imo was a bug.